.circleci/config.yml

```diff
@@ -17,33 +17,10 @@ jobs:
               echo "No changes to litellm or pyproject.toml. Skipping tests."
               circleci step halt
             fi
-      - restore_cache:
-          keys:
-            - v1-dependencies-{{ checksum ".circleci/requirements.txt" }}
       - run:
           name: Install Dependencies
           command: |
-            python -m pip install --upgrade pip
-            python -m pip install -r .circleci/requirements.txt
-            pip install "pytest==7.3.1"
-            pip install "pytest-asyncio==0.21.1"
-            pip install mypy
-            pip install "google-generativeai>=0.3.2"
-            pip install "google-cloud-aiplatform>=1.38.0"
-            pip install "boto3>=1.28.57"
-            pip install langchain
-            pip install "langfuse>=2.0.0"
-            pip install numpydoc
-            pip install traceloop-sdk==0.0.69
-            pip install openai
-            pip install prisma
-            pip install "httpx==0.24.1"
-            pip install "anyio==3.7.1"
-            pip install "asyncio==3.4.3"
-      - save_cache:
-          paths:
-            - ./venv
-          key: v1-dependencies-{{ checksum ".circleci/requirements.txt" }}
+            python3 -m poetry install
```

Comment:
Maybe you can use something like this to avoid the manual install of poetry:

```yaml
version: '2.1'
orbs:
  python: circleci/python@1.0.0
jobs:
  build:
    executor: python/default
    steps:
      - checkout
      - python/install-packages:
          pkg-manager: poetry
      - run:
          command: |
            poetry run pytest --version
          name: Test it
workflows:
  main:
    jobs:
      - build

```

as per https://circleci.com/developer/orbs/orb/circleci/python#usage-work-with-poetry

Reply:
tried that too, just see this obscure log 
```
Creating virtualenv litellm-3aSsmiER-py3.8 in /home/circleci/.cache/pypoetry/virtualenvs
Installing dependencies from lock file

Package operations: 51 installs, 0 updates, 0 removals


Exited with code exit status 1
```

any suggestions @bufferoverflow ?

Reply:
@ishaan-jaff  I'm pretty sure this is because of outdated poetry.lock file:

```sh
$ poetry install
Installing dependencies from lock file
Warning: poetry.lock is not consistent with pyproject.toml. You may be getting improper dependencies. Run `poetry lock [--no-update]` to fix it.

No dependencies to install or update

Installing the current project: litellm (1.16.19)
```

running `poetry lock --no-update` will update `poetry.lock` file and should fix this.

Reply:
hmm that did not fix the error, still seeing the same warning

Reply:
looks better already! according to https://app.circleci.com/pipelines/github/BerriAI/litellm/4413/workflows/e7ad0871-64b3-45d8-8b9b-8f5dd50b061c/jobs/6964 you need to add mypy to pyproject.toml as dev dependency using: `poetry add --group dev mypy`

Reply:
added - still battling a couple dependency conflicts